lib/vector-core/src/event/util/log/all_fields.rs

```diff
@@ -69,7 +69,7 @@ impl<'a> FieldsIter<'a> {
         loop {
             match path_iter.next() {
                 None => return res,
-                Some(PathComponent::Key(key)) => res.push_str(&key),
+                Some(PathComponent::Key(key)) => res.push_str(&key.replace(".", "\\.")),
```

Comment:
`str::replace` always allocates and copies into a new string. Would it make sense to check if the key contains `.` and skip the replace?

Reply:
Aha, I did not realize that but I guess yes, it would have to to return a `String` from a `&str`.

Reply:
Added the check in 47c8bd2

Reply:
WFM It would be nice if there were a set of `str` functions like that which returned `Cow` instead.